server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,278 @@
+import { Message, GuildTextableChannel, TextableChannel, User, EmbedOptions, Client } from 'eris'
+import { Command, DB } from '../imports/types'
+import fs from 'fs'
+import { promisify } from 'util'
+import { getInsult } from '../imports/tools'
+
+async function parseTriviaList (fileName: string) {
+  const readFile = promisify(fs.readFile)
+  const data = await readFile(`./triviaLists/${fileName}.txt`, 'utf8')
+  const triviaList = data.split('\n')
+  return triviaList
+    .map(el => el.replace('\n', '').split('`'))
+    .filter(el => el.length >= 2 && el[0])
+    .map(el => ({
+      question: el[0],
+      answers: el.slice(1).map(ans => ans.trim())
+    }))
+}
+
+export class TriviaSession {
+  revealMessages = ['I know this: ', 'Easy: ', 'Of course, it\'s: ']
+  failMessages = ['You suck at this', 'That was trivial, really', 'Moving on...']
+  settings: { maxScore: number, timeout: number, delay: number, botPlays: boolean, revealAnswer: boolean }
+  currentLine: {question: string, answers: string[]}
+  channel: TextableChannel
+  author: User
+  message: Message
+  questionList: {question: string, answers: string[]}[]
+  scores: { [char: string]: number } = {}
+  status = ''
+  timer: number
+  timeout: number
+  count: number
+  tempDB: DB
+  client: Client
+
+  constructor (triviaList: {question: string, answers: string[]}[], message: Message, botPlays: boolean, timeLimit: number, maxScore: number, revealAnswers: boolean, tempDB: DB, client: Client) {
+    this.channel = message.channel
+    this.author = message.author
+    this.message = message
+    this.questionList = triviaList
+    this.status = 'new question'
+    this.settings = { maxScore: maxScore, timeout: 120000, delay: timeLimit, botPlays: botPlays, revealAnswer: revealAnswers }
+    this.timer = null
+    this.count = 0
+    this.timeout = Date.now()
+    this.tempDB = tempDB
+    this.client = client
+  }
+
+  async sendScores () {
```

Comment:
Doesn't need to be async, and a bit of a misnomer so fixed that while I was at it.
```suggestion
  getScores () {
```